src/components/RescueDetails.tsx

```diff
@@ -0,0 +1,345 @@
+import {ArrowLeftIcon, CircleUser, HomeIcon, MapPinIcon, XIcon} from 'lucide-react'
+import {Card, CardContent, CardHeader, CardTitle} from '@/components/ui/card'
+import {Button} from '@/components/ui/button'
+import {Label} from '@/components/ui/label'
+import React, {useEffect, useState} from 'react'
+import {Badge} from '@/components/ui/badge'
+import Airtable from 'airtable'
+import Link from "next/link";
+
+export default function RescueDetails({id}: { id: string }) {
+    //state variables
+    const [birdRescue, setBirdRescue] = useState<BirdAlert | null>(null)
+    const [isLoading, setIsLoading] = useState(true)
+    const [error, setError] = useState<string | null>(null)
+    const [showAcceptForm, setShowAcceptForm] = useState(false)
+    const [rescuerName, setRescuerName] = useState('')
+    const [localRescuerName, setLocalRescuerName] = useState(rescuerName)
+    const [formError, setFormError] = useState<string | null>(null)
+    const [volunteers, setVolunteers] = useState<any[]>([])
+
+    //connecting to airtable
+    const airtable = new Airtable({apiKey: process.env.NEXT_PUBLIC_AIRTABLE_ACCESS_TOKEN})
+    const base = airtable.base(process.env.NEXT_PUBLIC_AIRTABLE_BASE_ID!)
+
+    const fetchBirdRescues = async () => {
```

Comment:
the code in this file is still very hackathon-y and could use some more refactoring in a future PR